### DISCUSSION TITLE
how to get appdir from rust code?

### DISCUSSION THREAD

**Discussion by dassio** [tauri-apps/tauri#5319] [Q&A] [answered]
for js there is an `appdir()` method, how can i get this in rust code?

**Comment by FabianLars** [comment, marked as answer]
https://docs.rs/tauri/latest/tauri/api/path/fn.app_dir.html or for easier use: https://docs.rs/tauri/latest/tauri/struct.PathResolver.html#method.app_dir
the latter can be used via an instance of App, AppHandle and Window - something like this:
```rs
let appdir = app_handle.path_resolver().app_dir().unwrap();
```

**Reply by dassio** [reply]
thanks, the hard part is i did not notice the auto inject of appHandler, i found your answer on discord :  https://tauri.app/v1/guides/features/command#accessing-an-apphandle-in-commands
